pkg/driver/driver_azure.go

```diff
@@ -99,6 +99,9 @@ func (d *AzureDriver) getVMParameters(vmName string, networkInterfaceReferenceID
 		tagList[idx] = to.StringPtr(element)
 	}
 
+	azureDataDisks := d.AzureMachineClass.Spec.Properties.StorageProfile.DataDisks
+	dataDisks := d.generateDataDisks(vmName, azureDataDisks)
```

Comment:
Do we need a `nil` and `len()` check here also just like in the case of Alicloud? Please check previous comment.

Reply:
done.